git.md

```diff
@@ -1,2 +1,139 @@
+* apply to shared commits (maybe after cleaning commits locally).
+
+## tl;dr
+
+* Shared commits should not be changed.
+* Changes in each shared commit should be closely related.
+
+## 1. Preserve shared history
+
+Shared commits should never change. Some Git commands preserve the history, but
+others don't. For example, `git commit --amend` and `git rebase --interactive`
+alter the git history; they help to overwrite  a messy commit history with
+a cleaner one; you should use them only before sharing your work. To cleanup
+shared history use instead `git revert`.  If you try to push altered history by
+accident, Git will stop you; to benefit from this protection  avoid `--force`
+(or `-f`) when pushing to shared repositories.
+
+See also:
+
+* [Cleaning _unshared_ history: Squashing all commits in a branch](https://youtu.be/08dhy3Zoob4).
+* [Cleaning _unshared_ history with `git rebase --interactive`](https://youtu.be/cMI8p1XhMzA).
+* [Cleaning _unshared_ history with `git commit --amend`](https://youtu.be/539pfVfr7OI).
+* [Cleaning shared history with `git revert`](https://youtu.be/A8Ld6iDqc3w).
+* [Git playlist](https://www.youtube.com/playlist?list=PLvgdJdJDL-AOHkwiaMvYhPKVjiD9vzZIo).
+
+## 2. Keep commits focused
+
+A commit should focus on related changes only. For example, fixing two
+different bugs should produce two separate commits. Small commits make it
+easier for other developers to understand the changes and roll them back if
+something went wrong. You can avoid sharing unwanted changes via .gitignore. If
+you can't succinctly describe what the commit is doing in a couple of words,
+then it is likely too big.
+
+See also:
+
+* [The right size of a PR](https://github.com/2DegreesInvesting/practices/discussions/3).
+* [.gitignore templates](https://github.com/github/gitignore).
+* Undoing things with Git ([video](https://youtu.be/dZOfEF19yDk),
+[lesson](https://coderefinery.github.io/git-intro/05-undoing)).
+
+## 3. Commit frequently
+
+Committing frequently keeps your commits small. Small commits are more likely
+to be focused; they are easier to review, and easier to merge because they
+offer fewer opportunities for conflicts.
+
+## 4. The work you share should be complete
+
+Each commit and feature should do what it promises, completely; and reverting
+it should undo what that commit or feature meant to do, also completely. If
+the goal of one commit requires changes in two files, that one commit should
+include changes to both files. Similarly, if the goal of a feature requires
+two kinds of changes, that feature should likely include two commits.
+
+## 5. Explain how to test the effect of the changes you made
+
+Testing your code is important, particularly when sharing it with
+others. Automated tests are the best but not the only way to test the effects
+of the changes you made. However you did it, explain how you ensured that the
+code behaves as you expect, and how someone else may reproduce your results.
```

Comment:
The instructions for testing should be included in either code or text files, not in git commit messages. commit messages are signposts to help other developers find the state of code at a point in time, not a forum for instruction or discussion.

Reply:
Thanks!

After 9dc246d the doc reads 
> When automated tests are excluded, the PR's first comment should
explain how you ensured that the code behaves as you expect,
and how someone else may reproduce your results.

The key phrase here is "the PR's first comment". While this is in line with your suggestion that the comment should not be in the git commit message, it is not exactly what you suggested -- that the comment should be in "either code or text file". Are you okay with this? If not, please suggest the bit of text you would propose (or expand on your idea as I didn't fully understand your suggestion about explaining the testing approach in text files).

Reply:
@AlexAxthelm while you are here, please confirm you are okay with this or suggest what you would like to read instead.